WordPress/openverse-catalog
# Refactor Metropolitan Museum of Art to use ProviderDataIngester

## Fixes
Fixes #641 by @obulat and #588 by @stacimc

## Description
This refactors the metropolitan museum to use the provider data ingester class, and hopefully improve search results by populating title and tags more reliably.

## Testing Instructions
When I was testing the dag in airflow, I kept hitting non-responses, until I set the delay at 3 seconds, rather than the standard 1 second. I'm not sure how sustainable that will be for production, since we need to hit the API once for each and every record, including non-CC0 records that we do not load. So, I do recommend testing out the dag in airflow, but with the rate limit caveat.

![image](https://user-images.githubusercontent.com/2520575/185264446-14d243a8-b626-4ee3-9ca4-5f3a62fdd2c0.png)

```
*Provider*: `metropolitan_museum`
*Date range*: 2022-08-08 -> 2022-08-09
*Duration of data pull task*: _No data_
*Number of records upserted per media type*:
  - `image`: 9,857 _(985 duplicate foreign IDs, 2,758 duplicate URLs)_

[2022-08-17, 17:46:31 UTC] {taskinstance.py:1395} INFO - Marking task as SUCCESS. dag_id=metropolitan_museum_workflow, task_id=report_load_completion, execution_date=20220808T000000, start_date=20220817T174631, end_date=20220817T174631
[2022-08-17, 17:46:31 UTC] {local_task_job.py:156} INFO - Task exited with return code 0
[2022-08-17, 17:46:31 UTC] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
```

And then go to `db-shell` and check that title and tags are always populated, try clicking through some of the urls, and so on. For example...

![image](https://user-images.githubusercontent.com/2520575/185264613-7a719e7c-02de-42ba-b990-c3c3ba6e2550.png)


## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
